src/types/decimal/magnitude.rs

```diff
@@ -0,0 +1,70 @@
+use crate::types::UInt;
+use num_bigint::{BigInt, BigUint};
+
+/// A simple wrapper type around a [`UInt`]. Users are not expected to construct
+/// instances of `Magnitude` directly; it is provided for conversion ergonomics.
+///
+/// Signed integer types cannot implement `Into<UInt>`. Instead, they implement
+/// `TryInto<UInt>` and report an error if the input integer is negative.
+///
+/// Signed integers can infallibly implement `Into<Magnitude>` by using their
+/// absolute value.
+pub struct Magnitude(UInt);
```

Comment:
🗺️ As mentioned above, this type was moved to its own module. It contains no new code apart from the `From<BigUint> for Magnitude` impl on line 32.